cclib/parser/serenityparser.py

```diff
@@ -32,11 +36,34 @@ def before_parsing(self):
         self.unrestricted = False
 
     def after_parsing(self):
+        # Get molecular orbital information
+        orbpath = self.path.parent / self.systemname / f"{self.systemname}.orbs.res.h5"
+        if os.path.isfile(orbpath):
+            assert find_package("h5py"), (
+                "h5py is needed to read in molecular orbital info from Serenity."
+            )
+            import h5py
+
+            with h5py.File(orbpath, "r") as orbfile:
+                coeffs = [orbfile["coefficients"][:]]
+                eigenvalues = [orbfile["eigenvalues"][:].flatten()]
+                self.set_attribute("moenergies", eigenvalues)
+                self.set_attribute("mocoeffs", coeffs)
+                self.set_attribute("nmo", len(eigenvalues[0]))
+
         super().after_parsing()
 
     def extract(self, inputfile, line):
         """Extract information from the file object inputfile."""
 
+        self.path = Path(inputfile.filenames[0]).resolve()
```

Comment:
Set this in `before_parsing`, otherwise it will be done on every call to `extract`.